### PR TITLE
LibJS: Add Array.prototype.keys()

### PR DESCRIPTION
**Pull request by davidot**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp [hunk 1]
```diff
@@ -1033,6 +1034,15 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::values)
     return ArrayIterator::create(global_object, this_object, Object::PropertyKind::Value);
 }
 
+JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::keys)
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp:1037]:
```suggestion
// 23.1.3.16 Array.prototype.keys ( ), https://tc39.es/ecma262/#sec-array.prototype.keys
JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::keys)
```

File: Userland/Libraries/LibJS/Tests/builtins/Array/Array.prototype.keys.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    const a = ["a", "b"];
+    const it = a.keys();
+    expect(it.next()).toEqual({ value: 0, done: false });
+    expect(it.next()).toEqual({ value: 1, done: false });
+    expect(it.next()).toEqual({ value: undefined, done: true });
+    a.push("c");
+    expect(it.next()).toEqual({ value: undefined, done: true });
+});
+
+test()
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/Array/Array.prototype.keys.js:46]: What's this for?!

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibJS/Tests/builtins/Array/Array.prototype.keys.js (hunk 2 above)
Comment (davidot) [reply Userland/Libraries/LibJS/Tests/builtins/Array/Array.prototype.keys.js]: Yeah not sure what happened there

**Comment by davidot** [comment]
Fixed the comments and squashed the commits. I think it is ready for merging

**Pull request merged**